File: rust/src/model/saetze/LaborDatenpaketAbschluss.rs
#![allow(dead_code, unused_imports, non_camel_case_types)]

use crate::model::Kontext::Kontext;
use crate::model::enums::Satzart::Satzart;
use crate::model::saetze::Satz::Satz;

/// Satzart: L (Labor)-Datenpaket-Abschluss "8221"
pub struct LaborDatenpaketAbschluss {
    #[feld(Value = "9300", Feldart = libldt3.annotations.Feldart.muss)]
    pruefsumme: String
}

impl Satz for LaborDatenpaketAbschluss {
}

impl Kontext for LaborDatenpaketAbschluss {
}

File: cs/libldt3/model/saetze/LaborDatenpaketAbschluss.cs
/*
 * Copyright 2016-2024  Christoph Brill <opensource@christophbrill.de>
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
using libldt3.attributes;
using libldt3.model;
using libldt3.model.enums;
using libldt3.model.regel.erlaubt;

namespace libldt3
{
    namespace model
    {
        namespace saetze
        {
            /// <summary>
            /// Satzart: L (Labor)-Datenpaket-Abschluss "8221"
            /// </summary>
            [Datenpaket(Value = Satzart.LaborDatenpaketAbschluss)]
            public class LaborDatenpaketAbschluss : Satz, Kontext
            {
                [Feld(Value = "9300", Feldart = Feldart.muss)]
                [Regelsatz(Value = new[] { typeof(E157) }, Laenge = 40)]
                public string Pruefsumme;

            }
        }
    }
}
